helm/idol-nifi/templates/nifi/nifi.yaml

```diff
@@ -0,0 +1,204 @@
+          - name: NIFI_CLUSTER_NODE_ADDRESS
+            value: $(POD_NAME).nifi.$(POD_NAMESPACE).svc.cluster.local # Use pod fqdn as node address
+          - name: NIFI_REMOTE_INPUT_SOCKET_HOST
+            value: $(POD_NAME).nifi.$(POD_NAMESPACE).svc.cluster.local # Use pod fqdn as input socket address
+          - name: NIFI_REMOTE_INPUT_HOST
+            value: $(POD_NAME).nifi.$(POD_NAMESPACE).svc.cluster.local # Use pod fqdn as input host address
+          - name: HOSTNAME
+            value: $(POD_IP) # Use pod ip as hostname
+          - name: NODE_IDENTITY
+            value: $(POD_NAME) # Use pod name as identity
+        envFrom:
+          - configMapRef:
+              name: nifi-cm
+              optional: false
+        volumeMounts:
+          - name: data
+            mountPath: /opt/nifi/nifi-current/data
+            subPath: data
+            readOnly: false
+          - name: data
+            mountPath: /opt/nifi/nifi-current/keytool
+            subPath: keytool
+            readOnly: false
+          - name: scripts
+            mountPath: /scripts
+            readOnly: false
+        lifecycle:
+          postStart:
+            exec:
+              command:
+              - /bin/bash
+              - -c
+              - |
+                logfile=/opt/nifi/nifi-current/logs/post-start.log
+                /scripts/security.sh | tee -a ${logfile}
+                /scripts/wait.sh | tee -a ${logfile}
+                /scripts/connect-registry.sh | tee -a ${logfile}
+                echo [$(date)] postStart completed | tee -a ${logfile}
+          preStop:
+            exec:
+              command:
+              - sh
+              - -c
+              - |
+                host=nifi-0.${NIFI_WEB_HTTP_HOST#*.}
+                port=$NIFI_WEB_HTTP_PORT
+                cluster_node_id=`curl -s http://$NIFI_WEB_HTTP_HOST:$NIFI_WEB_HTTP_PORT/nifi-api/controller/cluster | jq -r ".cluster.nodes[] | select(.address==\"$NIFI_WEB_HTTP_HOST\") | .nodeId "`
+
+                setAndWaitForStatus () {
+                    curl -s -X PUT http://$host:$port/nifi-api/controller/cluster/nodes/$cluster_node_id -H 'Content-Type: application/json' -d "{\"node\":{\"nodeId\":\"$cluster_node_id\",\"status\": \"$1\"}}"
+                    cluster_node_status="$1"
+                    while [ "$cluster_node_status" != "$2" ]
+                    do
+                        sleep 1
+                        cluster_node_status=`curl -s http://$host:$port/nifi-api/controller/cluster/nodes/$cluster_node_id | jq .node.status -r`
+                    done
+                }
+
+                if [ ! -z "$cluster_node_id" ]
+                then
+                    setAndWaitForStatus "DISCONNECTING" "DISCONNECTED"
+                    setAndWaitForStatus "OFFLOADING" "OFFLOADED"
+                    curl -s -X DELETE http://$host:$port/nifi-api/controller/cluster/nodes/$cluster_node_id
+                fi
```

Comment:
Worth putting this into a separate .sh file in the `scripts` mount?

Reply:
[8f12d06](https://github.com/opentext-idol/idol-containers-toolkit/pull/12/commits/8f12d06cfce23e46b00d89a6623ee25d4c4664c3)